### PR TITLE
Firehose support

### PR DESCRIPTION
**Pull request by viniciusccarvalho**
Due too many rebase conflicts with the old PR at (https://github.com/spring-cloud/spring-cloud-stream-modules/pull/3) I'm submitting a new PR to address the support for a firehose source.

### CODE REVIEW

File: firehose-source/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+Spring Cloud Stream Firehose Sample
+=============================
```
Comment (ericbottard) [inline firehose-source/README.md:2]: Could you rephrase this README to have it conform to _e.g._ this template: https://github.com/spring-cloud/spring-cloud-stream-modules/blob/c3d654c2a635474a3f160f62bca0460f91864c0c/counter-sink/README.adoc

Avoid the use of "sample", and simplify the instructions to the maximum.
Use of asciidoc vs markdown would be prefered too.

File: firehose-source/pom.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    <parent>
+        <artifactId>spring-cloud-stream-modules</artifactId>
+        <groupId>org.springframework.cloud.stream.module</groupId>
+        <version>1.0.0.BUILD-SNAPSHOT</version>
+    </parent>
+
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (ericbottard) [inline firehose-source/pom.xml]: This can be omitted because inherited

File: firehose-source/pom.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        <start-class>org.springframework.cloud.stream.module.firehose.FirehoseApplication</start-class>
+        <java.version>1.8</java.version>
```
Comment (ericbottard) [inline firehose-source/pom.xml]: Can this build with java7? if so, this can be omitted because inherited

File: firehose-source/pom.xml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    </properties>
+
+    <pluginRepositories>
```
Comment (ericbottard) [inline firehose-source/pom.xml]: Please remove this repo

File: pom.xml [hunk 5]
```diff
@@ -59,6 +60,35 @@
 			</plugin>
 		</plugins>
 	</build>
+	<!--
+We need to enable this profile for now in order to override the imports execution that removes any dependencies from google.
+
+As of now, the imports activation of spring-cloud-build is not working properly. It activates regardless of the inclusion of the .google file on the source folder.
+-->
+	<profiles>
```
Comment (ericbottard) [inline pom.xml:68]: Can this move down to the firehose pom?

File: firehose-source/src/main/proto/events/envelope.proto [hunk 6]
```diff
@@ -0,0 +1,50 @@
+package events;
```
Comment (ericbottard) [inline firehose-source/src/main/proto/events/envelope.proto]: I see you've removed the protobuf build part from the pom. So is versioning those `.proto` files.

Could you add instructions on how to regenerate the `.java` files from the `.proto` files? In a README in `src/main/proto` for example? Or directly in the top README.
Exact command-line options would be welcome, so that we make sure we can rebuild 2 years from now ... :)

File: pom.xml (hunk 5 above)
Comment (viniciusccarvalho) [inline pom.xml:68]: Nope, it won't work. I tried that. If its on the firehose pom, build will work on the local module, but will fail from the parent pom

File: firehose-source/src/main/java/org/springframework/cloud/stream/module/firehose/source/ByteBufferMessageConverter.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    public Object fromMessage(Message<?> message, Class<?> targetClass) {
+        ByteBuffer buffer = (ByteBuffer) message.getPayload();
+        byte[] bytes = new byte[buffer.remaining()];
+        buffer.get(bytes, 0, bytes.length);
+        Tuple result = TupleBuilder.tuple().build();
+        try {
+            EventFactory.Envelope envelope = EventFactory.Envelope.parseFrom(bytes);
+            result = TupleFactory.createTuple(envelope);
+        } catch (InvalidProtocolBufferException e) {
+            e.printStackTrace();
```
Comment (ericbottard) [inline firehose-source/src/main/java/org/springframework/cloud/stream/module/firehose/source/ByteBufferMessageConverter.java:57]: This needs to be better propagated / logged

**Comment by ericbottard** [comment]
Closing this in favor of https://github.com/spring-cloud/spring-cloud-stream-modules/pull/33

Let's continue discussion there
